app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,57 @@
+class FoodsController < ApplicationController
+  before_action :set_food, only: %i[show edit update destroy]
+
+  def index
+    @foods = Food.all
+  end
+
+  def new
+    @food = Food.new
+  end
+
+  def show; end
+
+  def edit; end
+
+  def create
+    @food = current_user.foods.build(food_params)
+    # @food = @user.food.new(food_params)
+
+    respond_to do |format|
+      if @food.save
+        format.html { redirect_to user_foods_path(current_user), notice: 'Food was successfully created.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /foods/1 or /foods/1.json
+  def update
+    respond_to do |format|
+      if @food.update(food_params)
+        format.html { redirect_to user_foods_path(current_user), notice: 'Food was successfully updated.' }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @food.destroy
+
+    respond_to do |format|
+      format.html { redirect_to user_foods_path(current_user), notice: 'Food was successfully destroyed.' }
+    end
+  end
+
+  private
+
+  def set_food
+    @food = Food.find(params[:id])
+  end
+
+  def food_params
+    params.require(:food).permit(:name, :measurement_unit, :price, :quantity)
+  end
+end
```

Comment:
Your food_controller is not behaving as it should. I have created [this ](https://www.loom.com/share/c15d089a799e485bab9efb56c61429dd?sid=15602e16-d36e-46d1-a72b-5eac8750963a)video where i explained what the issue is. Any food item that any user adds is accessible to all the users and any user can delete it. That should not be the case.

Reply:
well resolved! ✔️